src/init/AppRunner.ts

```diff
@@ -56,97 +50,77 @@ export class AppRunner {
+      .usage('node ./bin/server.js [args]')).parseSync();
 
-    // Gather settings for instantiating the server
-    const loaderProperties: IComponentsManagerBuilderOptions<App> = {
+    // Gather settings for instantiating VarResolver
+    const loaderProperties = {
       mainModulePath: resolveAssetPath(params.mainModulePath),
       dumpErrorState: true,
       logLevel: params.loggingLevel as LogLevel,
     };
-    const configFile = resolveAssetPath(params.config);
 
-    // Create and execute the app
-    this.createApp(loaderProperties, configFile, params)
-      .then(
-        async(app): Promise<void> => app.start(),
-        (error: Error): void => {
-          // Instantiation of components has failed, so there is no logger to use
-          stderr.write(`Error: could not instantiate server from ${configFile}\n`);
-          stderr.write(`${error.stack}\n`);
-          process.exit(1);
-        },
-      ).catch((error): void => {
-        this.logger.error(`Could not start server: ${error}`, { error });
-        process.exit(1);
-      });
+    // Create varResolver
+    const varConfigFile = resolveAssetPath(params.varConfig);
+    this.createComponent(
+      loaderProperties as IComponentsManagerBuilderOptions<VarResolver>, varConfigFile, {}, varConfigComponentIri,
+    ).then(
+    //  Using varResolver resolve vars and start app
+      async(varResolver): Promise<void> => {
+        let vars: VarRecord;
+        try {
+          vars = await varResolver.handle(argv);
+        } catch (error: unknown) {
+          this.exitWithError(error as Error, 'Error in computing variables', stderr);
+        }
+        return this.initApp(
+          loaderProperties, resolveAssetPath(params.config), vars as unknown as Record<string, any>, stderr,
+        );
+      },
+
+      (error): void => this.exitWithError(
+        error, `Error in loading variable configuration from ${varConfigFile}`, stderr,
+      ),
+    ).catch((error): void => this.exitWithError(error, 'Could not start the server', stderr));
   }
 
-  /**
-   * Creates the main app object to start the server from a given config.
-   * @param loaderProperties - Components.js loader properties.
-   * @param configFile - Path to a Components.js config file.
-   * @param variables - Variables to pass into the config file.
-   */
-  public async createApp(
+  private exitWithError(error: Error, message: string, stderr: WriteStream): never {
+    stderr.write(`message\n`);
+    stderr.write(`${error.stack}\n`);
+    process.exit(1);
+  }
+
+  private async initApp(
     loaderProperties: IComponentsManagerBuilderOptions<App>,
-    configFile: string,
-    variables: CliParams | Record<string, any>,
-  ): Promise<App> {
-    // Translate command-line parameters if needed
-    if (typeof variables.loggingLevel === 'string') {
-      variables = this.createVariables(variables as CliParams);
+    configFile: string, vars: Record<string, any>, stderr: WriteStream,
```

Comment:
```suggestion
    configFile: string,
    variables: Record<string, any>,
    stderr: WriteStream,
```
(and `stderr` should no longer be needed when we just throw errors)